WordPress/gutenberg
# Do not render global styles coming from the server in the site editor

## What?

This PR makes the global styles coming from the server not to be rendered in the site editor.

## Why?

The site editor, unlike other editors, generates the styles coming from global styles (presets, styles) dynamically in the client. This is because the user can modify any value that we need to consolidate (merge core, theme, and user style preferences). We don't want to have the global styles (coming from core or the plugin) twice, as it introduces subtle bugs.

## How?

This PR modifies how things work:

- the site editor receives now the same info than any other editor receives
- the styles that need to be removed are "tagged" with the `isGlobalStyles: true`

The styles sent by the server to the client end up in the `block-editor` store, to be found at `settings.styles`. By doing this, we leverage the existing mechanism in the site editor to recalculate the global styles upon user changes.

These are the styles before this PR (in WordPress 5.9 and WordPress trunk):

![Captura de ecrã de 2022-04-12 14-28-20](https://user-images.githubusercontent.com/583546/162963190-c96f67e7-c8ca-4dd2-8c7b-579781b72bc5.png)

And after this PR:

![Captura de ecrã de 2022-04-12 14-25-46](https://user-images.githubusercontent.com/583546/162963163-416ed0fa-5e30-45f6-b45c-5c6f114bc31e.png)

Note how the first three styles are no longer in the store.

## Testing Instructions

Check the block editor store:

- Use WordPress 5.8. Verify that the block editor store has only 3 styles before and after this PR.
- Use WordPress 5.9. Verify that the block editor store has only 3 styles after this PR (would have 6 before).
- Use this branch from wordpress-develop https://github.com/WordPress/wordpress-develop/pull/2542 Verify that the block editor store has only 3 styles after this PR (would have 6 before).

Do some random manual test:

- Add some block and set some preset values (colors, font sizes) as block styles and make sure they still work. Set some global styles and verify they work.
